sympy/external/tests/test_autowrap.py

```diff
@@ -308,4 +310,22 @@ def test_ufuncify_numpy():
     # This test doesn't use Cython, but if Cython works, then there is a valid
     # C compiler, which is needed.
     has_module('Cython')
-    runtest_ufuncify('C99', 'numpy')
+
+    # The ufuncify function generates a setup.py that uses distutils which
+    # is now deprecated:
+    #
+    # DeprecationWarning: The distutils package is deprecated and slated
+    # for removal in Python 3.12. Use setuptools or check PEP 632 for
+    # potential alternatives
```

Comment:
Places where distutils is used:
```console
$ git grep distutils
bin/test_travis.sh:import distutils.sysconfig
bin/test_travis.sh:with open(os.path.join(distutils.sysconfig.get_python_lib(), 'coverage.pth'), 'w') as pth:
doc/src/modules/utilities/autowrap.rst:system installed f2py command. The Cython backend runs a distutils setup script
release/fabfile.py:    distutils_check()
release/fabfile.py:def distutils_check():
setup.py:This uses Distutils (https://python.org/sigs/distutils-sig/) the standard
setup.py:from distutils.command.sdist import sdist
setup.py:    from distutils.core import setup, Command
setup.py:    user_options = []  # distutils complains if this is not here.
setup.py:    def initialize_options(self):  # distutils wants this
setup.py:    user_options = []  # distutils complains if this is not here.
setup.py:    def initialize_options(self):  # distutils wants this
setup.py:    user_options = []  # distutils complains if this is not here.
setup.py:    def initialize_options(self):  # distutils wants this
sympy/external/importtools.py:    # Simplified from distutils.version.LooseVersion which was deprecated in
sympy/utilities/_compilation/compilation.py:    # from distutils/command/build_ext.py:
sympy/utilities/_compilation/util.py:    # distutils: language = c++
sympy/utilities/autowrap.py:    from distutils.core import setup
sympy/utilities/autowrap.py:    from distutils.extension import Extension
sympy/utilities/autowrap.py:        The following optional parameters get passed to ``distutils.Extension``
sympy/utilities/autowrap.py:    from numpy.distutils.misc_util import Configuration
sympy/utilities/autowrap.py:    from numpy.distutils.core import setup
sympy/utilities/tests/test_autowrap.py:    from distutils.core import setup
sympy/utilities/tests/test_autowrap.py:    from distutils.extension import Extension
sympy/utilities/tests/test_autowrap.py:    from distutils.core import setup
sympy/utilities/tests/test_autowrap.py:    from distutils.extension import Extension
sympy/utilities/tests/test_autowrap.py:    from distutils.core import setup
sympy/utilities/tests/test_autowrap.py:    from distutils.extension import Extension
```
- Travis and fabfile don't matter.
- The main setup.py should be removed and replaced with `pyproject.toml`.
- The external module only shows up because of a comment.
- The problematic cases are all in autowrap.

The autowrap module needs to be changed to not use distutils or numpy.distutils which are both deprecated. It isn't immediately clear to me what the appropriate replacements for these are for a nontrivial setup.py that actually needs to compile some C code.